consultation_analyser/support_console/ingest.py

```diff
@@ -57,150 +56,145 @@ def get_themefinder_outputs_for_question(
     return json.loads(response["Body"].read())
 
 
-def import_themes(question_part: QuestionPart, theme_data: dict) -> Framework:
-    theme_generation_execution_run = ExecutionRun.objects.create(
-        type=ExecutionRun.TaskType.THEME_GENERATION
-    )
-    framework = Framework.create_initial_framework(
-        question_part=question_part, execution_run=theme_generation_execution_run
-    )
-    for theme_key, theme_value in theme_data.items():
-        name, description = theme_value.split(": ", 1)
-        logger.info(f"Creating theme: {name}, key: {theme_key}")
-        Theme.create_initial_theme(
-            framework=framework, key=theme_key, name=name, description=description
+def import_respondent_data(consultation: Consultation, respondent_data: list):
+    logger.info(f"Importing respondent data for consultation: {consultation.title}")
+    respondents_to_save = []
+    for respondent in respondent_data:
+        respondent = json.loads(respondent.decode("utf-8"))
+        themefinder_respondent_id = respondent["themefinder_id"]
+        # TODO - add further fields e.g. user supplied ID
+        respondents_to_save.append(
+            Respondent(
+                consultation=consultation, themefinder_respondent_id=themefinder_respondent_id
+            )
         )
-    return framework
-
-
-def get_theme_for_key(framework: Framework, key: str) -> Theme:
-    return Theme.objects.get(framework=framework, key=key)
-
-
-def create_answer_from_dict(
-    theme_mapping_dict: dict, question_part: QuestionPart, respondent: Respondent
-) -> Answer:
-    text = theme_mapping_dict["response"]
-    answer = Answer.objects.create(question_part=question_part, respondent=respondent, text=text)
-    return answer
-
+    Respondent.objects.bulk_create(respondents_to_save)
+    logger.info(f"Saved batch of respondents for consultation: {consultation.title}")
+
+
+def import_question_part_data(consultation: Consultation, question_part_dict: dict) -> QuestionPart:
+    type_mapping = {
+        "free_text": QuestionPart.QuestionType.FREE_TEXT,
+        "single_option": QuestionPart.QuestionType.SINGLE_OPTION,
+        "multiple_option": QuestionPart.QuestionType.MULTIPLE_OPTIONS,
+    }
+
+    question_text = question_part_dict.get("question_text", "")
+    question_part_text = question_part_dict.get("question_part_text", "")
+    if not question_text and not question_part_text:
+        raise ValueError("There is no text for question or question part.")
+
+    question_number = question_part_dict["question_number"]
+    question_part_number = question_part_dict.get(
+        "question_part_number", 1
+    )  # If no question_part_number, assume 1
+    question_part_type = question_part_dict.get("question_part_type", "free_text")
+    question_part_type = type_mapping[question_part_type]
+
+    question, _ = Question.objects.get_or_create(
+        consultation=consultation, number=question_number, defaults={"text": question_text}
```

Comment:
Is there a risk of errors if we're given different question_text values for the same question_number in different rows?

Reply:
Now will fail if the text changes (tested in an updated test).
https://github.com/i-dot-ai/consult/pull/515/commits/113c8361fdcfed5b9c035a9924177dcf04907c4a